airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeSqlOperations.java

```diff
@@ -73,4 +80,12 @@ protected String generateFilesList(final List<String> files) {
     }
   }
 
+  @Override
+  protected Optional<ConfigErrorException> checkForKnownConfigExceptions(Exception e) {
+    if (e instanceof SnowflakeSQLException && e.getMessage().contains(NO_PRIVILEGES_ERROR_MESSAGE)) {
+      return Optional.of(new ConfigErrorException("Encountered Error with Snowflake Configuration, please verify your privileges", e));
```

Comment:
can we be more specific about what permissions are needed? (this is an actual question, I don't know how specific snowflake's error message is)

Reply:
`+1` to Ed's comment, do we know what are the ways a customer can resolve this 

```
Schema 'PUBLIC' already exists, but current role has no privileges on it. If this is unexpected and you cannot resolve this problem, contact your system administrator.
```

Also, looking at the `SnowflakeSQLException` class, it extends off of `SQLException` which has a method `getSQLState()`, is that something we can more accurately map so this test doesn't become fragile?

Reply:
The SQL state was `SQLState = "42710"` but AFAICT this didn't map to anything beside the `SQLComilationError` here's the entire content of the exception 
```
this = {SnowflakeInternalStagingSqlOperations@5796} 
e = {SnowflakeSQLException@5797} "net.snowflake.client.jdbc.SnowflakeSQLException: SQL compilation error:\nSchema 'PUBLIC' already exists, but current role has no privileges on it. If this is unexpected and you cannot resolve this problem, contact your system administrator. ACCOUNTADMIN role may be required to manage the privileges on the object."
  queryId = "01a9da10-0000-eef5-0000-53b903d214ba"
  retryCount = 0
  issocketTimeoutNoBackoff = false
  elapsedSeconds = 0
  SQLState = "42710"
  vendorCode = 3041
  next = null
  backtrace = {Object[6]@5801} 
  detailMessage = "SQL compilation error:\nSchema 'PUBLIC' already exists, but current role has no privileges on it. If this is unexpected and you cannot resolve this problem, contact your system administrator. ACCOUNTADMIN role may be required to manage the privileges on the object."
  cause = {SnowflakeSQLException@5797} "net.snowflake.client.jdbc.SnowflakeSQLException: SQL compilation error:\nSchema 'PUBLIC' already exists, but current role has no privileges on it. If this is unexpected and you cannot resolve this problem, contact your system administrator. ACCOUNTADMIN role may be required to manage the privileges on the object."
  stackTrace = {StackTraceElement[29]@5806} 
  depth = 29
  suppressedExceptions = {Collections$EmptyList@5804}  size = 0
```
I'm open to any thoughts here

Reply:
This is a little tricky since Snowflake has poor documentation on their SQLSTATE codes but it appears a few other DB documentation [Postgres](https://www.postgresql.org/docs/current/errcodes-appendix.html) and [IBM](https://www.ibm.com/docs/en/i/7.3?topic=codes-listing-sqlstate-values) both use `42710` for marking `duplicate object`

I'll leave it up to your discretion, although I _might_ consider if you go with a string check to have it 

```
ACCOUNTADMIN role may be required to manage the privileges on the object.
```

Since if someone were to see that within the code it makes a bit more sense what they _may_ need to do in order to resolve their issue. Granted the usage of the `ConfigErrorException` is for a actionable message for the user to resolve this configuration issue